tracee-ebpf/tracee/vmlinux/vmlinux-core.h

```diff
@@ -0,0 +1,639 @@
+typedef unsigned short __kernel_sa_family_t;
+
+struct in_addr {
+	__be32 s_addr;
+};
+
+struct sockaddr_in {
+	__kernel_sa_family_t sin_family;
+	__be16 sin_port;
+	struct in_addr sin_addr;
+};
+
+struct unix_sock {
+	struct unix_address *addr;
+};
+
+struct sockaddr_un {
+	__kernel_sa_family_t sun_family;
+	char sun_path[108];
+};
+
+struct unix_address {
+	int len;
+	struct sockaddr_un name[0];
+};
+
+struct ipv6_pinfo {
+	struct in6_addr saddr;
+	__be32 flow_label;
+};
+
+struct sockaddr_in6 {
+	unsigned short sin6_family;
+	__be16 sin6_port;
+	__be32 sin6_flowinfo;
+	struct in6_addr sin6_addr;
+	__u32 sin6_scope_id;
+};
+
+struct linux_binprm {
+	struct file *file;
+	int argc;
+	int envc;
+	const char *filename;
+};
+
+typedef __s64 time64_t;
+
+struct timespec64 {
+	time64_t tv_sec;
+	long int tv_nsec;
+};
+
+struct inode {
+	umode_t i_mode;
+	struct super_block *i_sb;
+	long unsigned int i_ino;
+	struct timespec64 i_ctime;
+};
+
+struct super_block {
+	dev_t s_dev;
+};
+
+struct mm_struct {
+	struct {
+		long unsigned int arg_start;
+		long unsigned int arg_end;
+		long unsigned int env_start;
+		long unsigned int env_end;
+	};
+};
+
+struct vfsmount {
+	struct dentry *mnt_root;
+};
+
+struct mount {
+	struct mount *mnt_parent;
+	struct dentry *mnt_mountpoint;
+	struct vfsmount mnt;
+};
+
+struct qstr {
+	union {
+		struct {
+			u32 hash;
+			u32 len;
+		};
+		u64 hash_len;
+	};
+	const unsigned char *name;
+};
+
+struct dentry {
+	struct dentry *d_parent;
+	struct qstr d_name;
+};
+
+struct user_namespace {
+	struct ns_common ns;
+};
+
+struct ipc_namespace {
+	struct ns_common ns;
+};
+
+struct net {
+	struct ns_common ns;
+};
+
+typedef __u32 __wsum;
+
+struct cgroup_namespace {
+	struct ns_common ns;
+};
+
+typedef __kernel_sa_family_t sa_family_t;
+
+struct sockaddr {
+	sa_family_t sa_family;
+};
+
+struct iovec {
+	void *iov_base;
+	__kernel_size_t iov_len;
+};
+
+enum bpf_map_type {
+	BPF_MAP_TYPE_UNSPEC = 0,
+	BPF_MAP_TYPE_HASH = 1,
+	BPF_MAP_TYPE_ARRAY = 2,
+	BPF_MAP_TYPE_PROG_ARRAY = 3,
+	BPF_MAP_TYPE_PERF_EVENT_ARRAY = 4,
+	BPF_MAP_TYPE_PERCPU_HASH = 5,
+	BPF_MAP_TYPE_PERCPU_ARRAY = 6,
+	BPF_MAP_TYPE_STACK_TRACE = 7,
+	BPF_MAP_TYPE_CGROUP_ARRAY = 8,
+	BPF_MAP_TYPE_LRU_HASH = 9,
+	BPF_MAP_TYPE_LRU_PERCPU_HASH = 10,
+	BPF_MAP_TYPE_LPM_TRIE = 11,
+	BPF_MAP_TYPE_ARRAY_OF_MAPS = 12,
+	BPF_MAP_TYPE_HASH_OF_MAPS = 13,
+	BPF_MAP_TYPE_DEVMAP = 14,
+	BPF_MAP_TYPE_SOCKMAP = 15,
+	BPF_MAP_TYPE_CPUMAP = 16,
+	BPF_MAP_TYPE_XSKMAP = 17,
+	BPF_MAP_TYPE_SOCKHASH = 18,
+	BPF_MAP_TYPE_CGROUP_STORAGE = 19,
+	BPF_MAP_TYPE_REUSEPORT_SOCKARRAY = 20,
+	BPF_MAP_TYPE_PERCPU_CGROUP_STORAGE = 21,
+	BPF_MAP_TYPE_QUEUE = 22,
+	BPF_MAP_TYPE_STACK = 23,
+	BPF_MAP_TYPE_SK_STORAGE = 24,
+	BPF_MAP_TYPE_DEVMAP_HASH = 25,
+	BPF_MAP_TYPE_STRUCT_OPS = 26,
+	BPF_MAP_TYPE_RINGBUF = 27,
+	BPF_MAP_TYPE_INODE_STORAGE = 28,
+	BPF_MAP_TYPE_TASK_STORAGE = 29,
+};
+
+struct bpf_map {
+	u32 id;
+	char name[16];
+};
+
+struct bpf_sock;
+
+// TODO: can't CO-RE __sk_buff (check)
```

Comment:
Can you please explain? Should we open an issue for that?

Reply:
Yes I'll open an issue and I think it's because of offset based pointers being used, but just a runch now.. have to ivestigate

Reply:
The following issue has been opened so we can address this subsequently: https://github.com/aquasecurity/tracee/issues/1172